src/webui/api/torrentscontroller.cpp

```diff
@@ -113,17 +118,84 @@ namespace
     void applyToTorrents(const QStringList &hashes, const std::function<void (BitTorrent::TorrentHandle *torrent)> &func)
     {
         if ((hashes.size() == 1) && (hashes[0] == QLatin1String("all"))) {
-            foreach (BitTorrent::TorrentHandle *torrent, BitTorrent::Session::instance()->torrents())
+            foreach (BitTorrent::TorrentHandle *const torrent, BitTorrent::Session::instance()->torrents())
                 func(torrent);
         }
         else {
             for (const QString &hash : hashes) {
-                BitTorrent::TorrentHandle *torrent = BitTorrent::Session::instance()->findTorrent(hash);
+                BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
                 if (torrent)
                     func(torrent);
             }
         }
     }
+
+    QVariantList getStickyTrackers(const BitTorrent::TorrentHandle *const torrent)
+    {
+        uint seedsDHT = 0, seedsPeX = 0, seedsLSD = 0, leechesDHT = 0, leechesPeX = 0, leechesLSD = 0;
+        for (const BitTorrent::PeerInfo &peer : torrent->peers()) {
+            if (peer.isConnecting()) continue;
+
+            if (peer.fromDHT()) {
+                if (peer.isSeed())
+                    ++seedsDHT;
+                else
+                    ++leechesDHT;
+            }
+            if (peer.fromPeX()) {
+                if (peer.isSeed())
+                    ++seedsPeX;
+                else
+                    ++leechesPeX;
+            }
+            if (peer.fromLSD()) {
+                if (peer.isSeed())
+                    ++seedsLSD;
+                else
+                    ++leechesLSD;
+            }
+        }
+
+        const QString working {QCoreApplication::translate("TrackerListWidget", "Working")};
```

Comment:
Well you will not convince me that this is the correct way! Haha, it depends on the translations from the GUI components!.. However, I don't want to open your eyes anymore so you can ignore this comment.

Reply:
Addressed this in my latest commit. Translation of `status` is now done client-side. The `msg` is still translated server-side since in most cases it's not actually translated at all. It might even make sense to remove translation of the `msg` field entirely.

Reply:
>It might even make sense to remove translation of the msg field entirely.

Real trackers can return untranslated messages so it doesn't make much sense to translate this message.
Otherwise, if you still want to do it, I wouldn't rely on translation from GUI and have own one.